ElectricityBillApi/Controllers/ElectricityProviderController.cs

```diff
@@ -0,0 +1,26 @@
+﻿using ElectricityBillApi.Models;
+using ElectricityBillApi.Services;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+
+namespace ElectricityBillApi.Controllers
+{
+    [ApiController]
+    [Route("api/v1/[controller]")]
+    public class ElectricityProviderController : ControllerBase
+    {
+        private readonly IElectricProviderPickerService _electricProviderPickerService;
+
+        public ElectricityProviderController(IElectricProviderPickerService electricProviderPickerService)
+        {
+            _electricProviderPickerService = electricProviderPickerService;
+        }
+
+        [HttpGet]
+        public IActionResult Get([FromBody] Address address)
+        {
+            var provider = _electricProviderPickerService.FindCheapest(address);
+            return Ok(new { Name = provider.Name, Price = provider.CalculatePrice(address)});
```

Comment:
I would move this: `Price = provider.CalculatePrice(address)}` to a different line, because it's a bit difficult to read. A line, just like a function- should do one ting.
In your case, it actually does 3 things:
returns Ok with a new object and calculates price.

Also, consider moving adding `FindCheapestProviderPrice` method to return strongly-typed cheapest provider and its price.

Reply:
Changed it to return a `Tuple(ElectricProvider, decimal)` with the price calculated (since I already calculated it anyways to find the provider)